lib/sqlalchemy/sql/schema.py

```diff
@@ -5664,3 +5707,239 @@ def _copy(self, **kw: Any) -> Identity:
         )
 
         return self._schema_item_copy(i)
+
+
+class Period(SchemaItem):
+    """Defines a period construction, i.e. ``PERIOD FOR period_name
+    (start_col, end_col)`` syntax.
+
+    .. versionadded:: TODO
+    """
+
+    __visit_name__ = "period"
+
+    table: Table
+    start: Column
+    end: Column
+    system: bool
+
+    def __init__(
+        self,
+        name: str,
+        start: _DDLColumnArgument,
```

Comment:
**Federico Caselli** (CaselIT) wrote:

maybe we could avoid this and have just period?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3803